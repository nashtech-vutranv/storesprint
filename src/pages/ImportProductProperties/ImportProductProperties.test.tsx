import {render, waitFor} from '@testing-library/react'
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'
import {ImportProductProperties} from './'

const tree = render(
    <MemoryRouter>
      <ImportProductProperties />
    </MemoryRouter>
  )

jest.mock('react-router-dom', () => {
  return {
    ...jest.requireActual('react-router-dom'),
    useParams: jest.fn(),
  }
})

describe('Import product properties Page', () => {
  it('should match snapshot with successful data from api', async () => {
    jest.mock('../../services/ImportProductPropertiesService', () => {
      const instance = jest.fn().mockReturnValue({
        importProductProperties: jest.fn().mockImplementation(() => {
          return Promise.resolve({
            data: {
              errorMessage: '',
              invalidProperties: [],
              success: 0,
              total: 0,
            },
          })
        }),
      })

      return instance
    })

    await waitFor(() => {
      expect(tree).toMatchSnapshot()
      expect(
        tree.queryByText(/import_product_properties_title/i)
      ).toBeInTheDocument()
      expect(
        tree.queryByText(/import_product_properties_choose_file/i)
      ).toBeInTheDocument()
      expect(
        tree.queryByText(/import_product_properties_allow_override/i)
      ).toBeInTheDocument()
      expect(tree.queryByText(/common_confirm_cancel/i)).toBeInTheDocument()
      expect(tree.queryByText(/common_submit/i)).toBeInTheDocument()
    })
  })

  it('should match snapshot with error from API', async () => {
    jest.mock('../../services/ImportProductPropertiesService', () => {
      const mockValue = jest.fn().mockReturnValue({
        importProductProperties: jest.fn().mockImplementation(() => {
          return Promise.reject()
        }),
      })

      return mockValue
    })

    await waitFor(() => {
      expect(tree).toMatchSnapshot()
    })
  })
})
