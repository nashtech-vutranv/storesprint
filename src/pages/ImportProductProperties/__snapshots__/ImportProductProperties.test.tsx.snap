// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import product properties Page should match snapshot with error from API 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body />,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Import product properties Page should match snapshot with successful data from api 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-form mt-3 card"
      >
        <div
          class="card-header"
        >
          <h4
            class="card-form__title"
          >
            import_product_properties_title
          </h4>
        </div>
        <div
          class="card-body"
        >
          <div
            class="d-flex align-items-center"
          >
            <button
              class="btn btn-success"
              type="button"
            >
              import_product_properties_choose_file
            </button>
            <div
              style="margin-left: 0.75rem; display: inline-block; max-width: 500px;"
            >
              <div />
              
            </div>
            <div
              class="invisible"
              style="margin-left: 1.5rem; width: 100%; max-width: 300px; transition: visibility ease-out 0.3s;"
            >
              <div
                class=" progress-sm progress"
              >
                <div
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuenow="0"
                  class="progress-bar bg-success progress-bar-animated progress-bar-striped"
                  role="progressbar"
                  style="width: 0%;"
                />
              </div>
            </div>
            <input
              accept="text/csv"
              class="d-none"
              type="file"
            />
          </div>
          <div
            class="d-flex mt-2"
          >
            <input
              id="allowOverride"
              name="allowOverride"
              type="checkbox"
            />
            <label
              class="d-inline-block"
              for="allowOverride"
              style="margin-left: 0.75rem;"
            >
              import_product_properties_allow_override
            </label>
          </div>
          <div
            class="row"
          >
            <div
              class="col-8 offset-4"
            >
              <button
                class="me-2 btn btn-danger"
                type="button"
              >
                common_confirm_cancel
              </button>
              <button
                class="me-2 btn btn-success"
                disabled=""
                type="button"
              >
                common_submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-form mt-3 card"
    >
      <div
        class="card-header"
      >
        <h4
          class="card-form__title"
        >
          import_product_properties_title
        </h4>
      </div>
      <div
        class="card-body"
      >
        <div
          class="d-flex align-items-center"
        >
          <button
            class="btn btn-success"
            type="button"
          >
            import_product_properties_choose_file
          </button>
          <div
            style="margin-left: 0.75rem; display: inline-block; max-width: 500px;"
          >
            <div />
            
          </div>
          <div
            class="invisible"
            style="margin-left: 1.5rem; width: 100%; max-width: 300px; transition: visibility ease-out 0.3s;"
          >
            <div
              class=" progress-sm progress"
            >
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="0"
                class="progress-bar bg-success progress-bar-animated progress-bar-striped"
                role="progressbar"
                style="width: 0%;"
              />
            </div>
          </div>
          <input
            accept="text/csv"
            class="d-none"
            type="file"
          />
        </div>
        <div
          class="d-flex mt-2"
        >
          <input
            id="allowOverride"
            name="allowOverride"
            type="checkbox"
          />
          <label
            class="d-inline-block"
            for="allowOverride"
            style="margin-left: 0.75rem;"
          >
            import_product_properties_allow_override
          </label>
        </div>
        <div
          class="row"
        >
          <div
            class="col-8 offset-4"
          >
            <button
              class="me-2 btn btn-danger"
              type="button"
            >
              common_confirm_cancel
            </button>
            <button
              class="me-2 btn btn-success"
              disabled=""
              type="button"
            >
              common_submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
